### PR TITLE
Add support for Mad Dog gm905

### PR DESCRIPTION
**Pull request by koliwbr**
https://maddog.pl/produkty/myszki/57,mad-dog-gm905

### CODE REVIEW

File: data/devices/sinowealth-0029.device
Comment (staticssleever668) [inline data/devices/sinowealth-0029.device:20]:
```suggestion
Buttons=14
DeviceName=Mad Dog GM905
LedType=RGB
SensorType=PMW3389
```

**Review by staticssleever668 — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution!
I changed the format of sinowealth device files for consistency with other drivers, so it needs to be slightly adjusted (that's why device file parse test fails).
Also, could you please attach the output of verbose ratbagd for your device? `sudo systemctl stop ratbagd.service && sudo ratbagd`.

**Comment by koliwbr** [comment]
`ratbagd --verbose`
```
Initializing libratbag
ratbag debug: New device: Corsair CORSAIR K55 RGB PRO Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1b1c:1ba4
ratbag debug: New device: Corsair CORSAIR K55 RGB PRO Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1b1c:1ba4
ratbag debug: New device: Corsair CORSAIR K55 RGB PRO Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1b1c:1ba4
ratbag debug: New device: SINOWEALTH Wired Gaming Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver sinowealth
ratbag debug: hidraw info: bus 0x03 vendor 0x258a product 0x29
ratbag debug: SINOWEALTH Wired Gaming Mouse is device '/dev/hidraw0'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x258a product 0x29
ratbag debug: SINOWEALTH Wired Gaming Mouse is device '/dev/hidraw5'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: - HID report ID 04
ratbag debug: - HID report ID 07
ratbag debug: - HID report ID 05
ratbag raw: feature set:   05 02 01 00 00 00
ratbag raw: feature set:   05 01 00 00 00 00
ratbag raw: feature get:   05 01 56 32 38 37
ratbag info: firmware version: V287
ratbag raw: feature set:   05 11 00 00 00 00
ratbag raw: feature get:   04 11 00 00 00 00 00 00 64 0f 03 38 00 08 10 14 14 14 14 14 14 00 00 00 00 00 00 00 00 ff 00 00 00 ff 00 00 00 ff ff ff 00 00 ff ff ff 00 ff ff ff ff ff 50 00 01 43 00 40 ff 00 ff 42 07 ff 00 00 00 ff 00 00 00 ff 00 ff ff ff ff 00 fa 00 fa ff ff ff 42 43 40 ff 69 b4 ff 69 b4 ff 00 00 ff 00 00 ff 69 b4 ff 69 b4 00 00 00 00 00 00 ff 00 00 ff 00 00 42 00 ff 00 ff 00 ff 42 43 02 ff 00 00 01 00
ratbag debug: Configuration size is 131 bytes
ratbag raw: feature set:   05 12 00 00 00 00
ratbag raw: feature get:   04 12 00 00 00 00 00 00 11 01 00 00 11 02 00 00 11 04 00 00 12 01 00 00 41 00 00 00 70 05 01 01 11 10 00 00 11 08 00 00 11 01 00 00 11 01 00 00 11 01 00 00 21 00 21 00 21 00 22 00 70 01 04 01 50 01 00 00 50 01 00 00 50 01 00 00 50 01 00 00 50 01 00 00 50 01 00 00
ratbag info: Found device: Mad Dog GM905
ratbag debug: Read button 5 activates macro 5: 0x1 0x1
ratbag debug: Read button 13 activates macro 1: 0x4 0x1
ratbag raw: feature set:   05 02 00 00 00 00
ratbag raw: feature get:   05 02 01 00 00 00
ratbag debug: Current profile index: 0
ratbag raw: feature set:   05 1a 00 00 00 00
ratbag raw: feature get:   05 1a 05 00 00 00
ratbag info: Debounce time: 10 ms
ratbag debug: driver match found: Sinowealth
hidraw0: "SINOWEALTH Wired Gaming Mouse", 1 profiles
ratbag debug: New device: SINOWEALTH Wired Gaming Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver sinowealth
ratbag debug: hidraw info: bus 0x03 vendor 0x258a product 0x29
ratbag debug: SINOWEALTH Wired Gaming Mouse is device '/dev/hidraw0'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
DBus server ready
```
`ratbagctl`
```
warbling-mara: SINOWEALTH Wired Gaming Mouse
```

**Comment by staticssleever668** [comment]
> I hope these hex values are not the contents of /etc/shadow (or PGP etc.) :)

Oh, no, it's just mouse memory. :)
The one starting with '04 11' is `struct sinowealth_config_report` in our code, '04 12' is `struct sinowealth_button_report`.
What I wanted is, configuration size, whether debounce time works well, and to have a copy of the memory of this mouse saved as a backup.

**Review by staticssleever668 — APPROVED** [review]
Tests fails unrelated (#1590). Thank you! Will merge when I have some more time.

Note to self, add this before merging:
```ini
# Configuration data size: 131 bytes.
# https://github.com/libratbag/libratbag/pull/1587
```
(you could commit that yourself, but I'm likely going to rebase your branch before merging anyway, so we can save us some bike shedding right now):

**Pull request merged**
